atom/browser/native_window_mac.mm

```diff
@@ -692,6 +693,21 @@ - (void)drawRect:(NSRect)dirtyRect {
   rwhv->ShowDefinitionForSelection();
 }
 
+void NativeWindowMac::SetVisibleOnAllWorkspaces(bool visible) {
+  NSUInteger collectionBehavior = [window_ collectionBehavior];
+  is_visible_on_all_workspaces_ = visible;
+  if (visible) {
+    collectionBehavior |= NSWindowCollectionBehaviorCanJoinAllSpaces;
+  } else {
+    collectionBehavior &= ~NSWindowCollectionBehaviorCanJoinAllSpaces;
+  }
+  [window_ setCollectionBehavior:collectionBehavior];
+}
+
+bool NativeWindowMac::IsVisibleOnAllWorkspaces() {
+  return is_visible_on_all_workspaces_;
```

Comment:
We should test whether `collectionBehavior` includes `NSWindowCollectionBehaviorCanJoinAllSpaces` instead of remembering the result.